syntax = "proto3";

package graphnexus;

// The request carries a single node ID.
message NeighborRequest {
  int32 node_id = 1;
}

// The response returns zero or more neighbors.
message NeighborResponse {
  repeated int32 neighbors = 1;
}

// Our service declaration.
service GraphService {
  // Given a node ID, returns its immediate neighbors.
  rpc GetNeighbors(NeighborRequest) returns (NeighborResponse);
}
